### ISSUE TITLE
--fix / Autofix removes one existing space from YAML comments and fails ansible-lint yaml[comments] afterwards

### ISSUE THREAD

**Issue by netopsengineer** [ansible/ansible-lint#3839]
##### Summary

When using the `--fix` option, the rules for `yaml[comment]` (which in the docs presently, does not mention is part of Autofix) mutates existing inline comments to single whitespace character, and subsequently fails the rules for `yaml[comment]`, which expects two whitespace characters.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 6.20.3 using ansible-core:2.15.5 ansible-compat:4.1.10 ruamel-yaml:0.17.35 ruamel-yaml-clib:0.2.8
```

- ansible installation method: `pip` @ `ansible-core==2.15.5`
- ansible-lint installation method: `pip`

##### STEPS TO REPRODUCE

```console (paste below)
# existing correct inline commenting
---
- name: Correct example
  hosts: localhost
  tasks:
    - name: Echo a message
      ansible.builtin.command: echo hello  # <-- command is better in this case
      changed_when: false

# after '--fix'
---
- name: Problematic example
  hosts: localhost
  tasks:
    - name: Echo a message
      ansible.builtin.shell: echo hello # <-- command is better in this case ( missing whitespace )
      changed_when: false
```

##### Desired Behavior

Only mutate inline `YAML` comments when required, to be conformant.

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

Please give some details of what is happening.
Include a [minimum complete verifiable example] with:

- minimized playbook to reproduce the error
- the output of running ansible-lint including the command line used
- if you're getting a stack trace, also the output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by netopsengineer** [comment]
Thanks for the quick turn around! @ssbarnea
